File: List/Find_kth_node.cpp
#include<bits/stdc++.h>
using namespace std;

struct node{
    int data;
    struct node *next;
};

typedef struct node *N;

N getnode(int x){
    N t=(struct node *)malloc(sizeof(struct node));
    t->data = x;
    t->next = NULL;
    return t;
}

N insert(N start,int x){
    N t = getnode(x);

    if(!start){
        return t;
    }

    N c = start;

    while(c->next!=NULL){
        c=c->next;
    }

    c->next = t;
    return start;
}

void display(N start){
    N t=start;

    while(t){
        cout << t->data << endl;
        t=t->next;
    }

}

int find_kth_node(N start,int k){
    N temp,kth_node;
    temp=start;
    kth_node = start;

    for(int i=0;i<k-1;i++){
        temp=temp->next;
    }

    while(temp->next!=NULL){
        temp = temp->next;
        kth_node=kth_node->next;
    }

    if(kth_node)
        return kth_node->data;

    return -1;
}

int main(){
    N start = NULL;

    for(int i=0;i<5;i++){
        start = insert(start, i);
    }

    display(start);

    int k;
    cout << "Enter the vaue for k :- ";
    cin >> k;
    cout << "Kth node from end is :- " << find_kth_node(start,k);

    return 0;
}

File: List/Find_kth_node.py
class node:
    def __init__(self, data):
        self.data = data
        self.next = None


class ListNode:

    def __init__(self):
        self.head = None

    def insert(self, val):
        if self.head is None:
            self.head = node(val)
        else:
            temp = self.head

            while temp.next is not None:
                temp = temp.next

            temp.next = node(val)

    def display(self):
        temp = self.head

        while temp is not None:
            print(temp.data)
            temp = temp.next

    def find_kth_node(self, k):
        tmp = self.head
        kth_node = self.head


        for i in range(0,k-1):
            tmp = tmp.next

        while tmp.next is not None:
            tmp = tmp.next
            kth_node = kth_node.next

        if kth_node:
            return kth_node.data
        else:
            return -1


if __name__ == "__main__":

    l = ListNode()

    for i in range(5):
        i+=1
        l.insert(i)

    l.display()

    k = int(input('Enter the value for k :- '))

    print(f'Value of kth node from end :- {l.find_kth_node(k)}')
